core/state/database.go

```diff
@@ -121,12 +121,20 @@ type cachingDB struct {
 
 // OpenTrie opens the main account trie at a specific root hash.
 func (db *cachingDB) OpenTrie(root common.Hash) (Trie, error) {
-	return trie.NewSecure(root, db.db)
+	tr, err := trie.NewSecure(root, db.db)
+	if err != nil {
+		return nil, err
+	}
+	return tr, nil
 }
 
 // OpenStorageTrie opens the storage trie of an account.
 func (db *cachingDB) OpenStorageTrie(addrHash, root common.Hash) (Trie, error) {
-	return trie.NewSecure(root, db.db)
+	tr, err := trie.NewSecure(root, db.db)
+	if err != nil {
+		return nil, err
+	}
+	return tr, nil
```

Comment:
Any particular reason to add these changes?

Reply:
Yes, there is in fact :) 
As it was previously, if if `trie.NewSecure` would return `nil, error.New(..)`, this method would return a non-nil Trie interface, where the interface Value would be `nil`. But the actual returned object would be a non-nil interface. 
These things have bitten us from time to time, as both @karalabe and @fjl can attest. And it bit me too, I can't remember exactly where now, but it prompted me to change it to return an actual `nil` nil, not a `interface with nil value` nil.

Reply:
oh damn, I see.